src/main/java/org/reaktivity/nukleus/maven/plugin/internal/generate/StringFlyweightGenerator.java

```diff
@@ -203,8 +203,8 @@ private MethodSpec setDirectBufferMethod()
                     .addParameter(DIRECT_BUFFER_TYPE, "srcBuffer")
                     .addParameter(int.class, "srcOffset")
                     .addParameter(int.class, "length")
-                    .addStatement("buffer().putByte(offset(), (byte) length)")
-                    .addStatement("buffer().putBytes(offset() + 1, srcBuffer, srcOffset, length)")
+                    .addStatement("buffer().putShort(offset(), (short) length, java.nio.ByteOrder.BIG_ENDIAN)")
```

Comment:
The order should just be native for shared memory since there is no advantage to making the "wrong" platforms do more work.